src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1364,28 +1364,24 @@ private IOperation FinishVisitingStatement(IOperation originalOperation, IOperat
             return result ?? MakeInvalidOperation(originalOperation.Syntax, originalOperation.Type, ImmutableArray<IOperation>.Empty);
         }
 
-        private void VisitStatements(ArrayBuilder<IOperation> statements)
-        {
-            foreach (var statement in statements)
-            {
-                VisitStatement(statement);
-            }
-        }
-
         private void VisitStatements(ImmutableArray<IOperation> statements)
         {
-            foreach (var statement in statements)
+            for (int i = 0; i < statements.Length; i++)
             {
-                VisitStatement(statement);
+                if (statements[i] is VariableDeclarationGroupOperation declarationGroup
+                    && declarationGroup.DeclarationKind != VariableDeclarationKind.Default)
+                {
+                    // visit the using decl with the following statements
+                    var followingStatements = ImmutableArray.Create(statements, Math.Min(i + 1, statements.Length - 1), statements.Length - i - 1);
```

Comment:
>Math.Min(i + 1, statements.Length - 1) [](start = 80, length = 38)

Won't this copy the last statement in the block, even if that is the `using` statement?

Reply:
`ImmutableArray.Create(statements, i + 1, statements.Length - i)` should work, even if `i + 1 == statements.Length`.

---
In reply to: [335196847](https://github.com/dotnet/roslyn/pull/39125#discussion_r335196847) [](ancestors = 335196847)

Reply:
No, because the length becomes zero in that case and you end up with an empty array. I've added `UsingDeclaration_Flow_11` to test this scenario.

Reply:
But what about when there is only one statement in the list, which is a using decl? `i = 0 => i + 1 = 1, statements.Length - i == 1` which is incorrect. 

---
In reply to: [335199463](https://github.com/dotnet/roslyn/pull/39125#discussion_r335199463) [](ancestors = 335199463,335196847)

Reply:
No, because the length becomes zero in that case and you end up with an empty array. I've added `UsingDeclaration_Flow_11` to test this scenario.

---
In reply to: [335196847](https://github.com/dotnet/roslyn/pull/39125#discussion_r335196847) [](ancestors = 335196847)

Reply:
Discussed offline + fixed

---
In reply to: [335205220](https://github.com/dotnet/roslyn/pull/39125#discussion_r335205220) [](ancestors = 335205220,335196847)